junit-runtime/src/main/scala/org/scalajs/junit/JUnitExecuteTest.scala

```diff
@@ -64,147 +73,145 @@ final class JUnitExecuteTest(taskDef: TaskDef, runner: JUnitBaseRunner,
       method: JUnitMethodMetadata) = {
     val jUnitMetadata = classMetadata.metadata()
     val testClassInstance = classMetadata.newInstance()
+    val methodName = method.name
+    val decodedMethodName = {
+      if (decodeScalaNames) runner.runSettings.decodeName(methodName)
+      else methodName
+    }
+    val testAnnotation = method.getTestAnnotation.get
 
     val t0 = System.nanoTime
-    def timeInSeconds() = (System.nanoTime - t0).toDouble / 1000000000
+    def getTimeInSeconds(): Double = (System.nanoTime - t0).toDouble / 1000000000
 
-    val beforeMethodsAssertionFaild = {
+    def executeTestMethods(): Unit = {
+      val expectedException = testAnnotation.expected
       try {
-        for (method <- jUnitMetadata.beforeMethod)
-          classMetadata.invoke(testClassInstance, method.name)
-        false
+        if (verbose) logFormattedInfo(decodedMethodName, "started")
+        else logFormattedDebug(decodedMethodName, "started")
+
+        classMetadata.invoke(testClassInstance, method.name)
+        logFormattedDebug(decodedMethodName,
+            s"finished, took ${getTimeInSeconds()} sec")
+
+        if (expectedException == classOf[org.junit.Test.None]) {
+          testPassed(methodName)
+          executeAfterMethods()
+        } else {
+          val msg = {
+            s"failed: Expected exception: " + expectedException +
+                s"took ${getTimeInSeconds()} sec"
+          }
+          logFormattedError(decodedMethodName, msg, None)
+          testFailed(methodName)
+        }
       } catch {
-        case ex: AssumptionViolatedException =>
-          logFormattedInfo(method.name, "started")
-          logAssertionWarning(method.name, ex, timeInSeconds())
-          true
-
-        case ex: internal.AssumptionViolatedException =>
-          logFormattedInfo(method.name, "started")
-          logAssertionWarning(method.name, ex, timeInSeconds())
-          true
+        case ex: Throwable =>
+          val timeInSeconds = getTimeInSeconds()
+          if (classOf[AssumptionViolatedException].isInstance(ex) ||
+              classOf[internal.AssumptionViolatedException].isInstance(ex)) {
```

Comment:
Why not `ex.isInstanceOf[AssumptionViolatedException] || ex.isInstanceOf[internal.AssumptionViolatedException`?